### ISSUE TITLE
Failure to click a button when calling getElement multiple times

### ISSUE THREAD

**Issue by Rekubot** [appium/java-client#645]
## Description

I have a collection of tests that locate the same button via accessibility ID, a couple of which attempt to click it. Running the tests in isolation work fine, but when run in sequence I get an error when calling the click method. I imagine I can work around it but wanted to raise the issue anyway.

## Environment

* java client build version : 5.0.0-BETA8
* Appium server version: v1.6.5-beta
* Node.js version (unless using Appium.app|exe) or Appium CLI or Appium.app|exe: 7.10.0
* Mobile platform/version under test: iOS 10.3
* Real device or emulator/simulator: Simulator

## Exception stacktraces

[Xcode] 2017-05-19 17:07:10.896 XCTRunner[25815:242886] Enqueue Failure: Failure fetching attributes for element <XCAccessibilityElement: 0x608000a55a50> Device element: Error Domain=XCTDaemonErrorDomain Code=13 "Error copying attributes -25202" UserInfo={NSLocalizedDescription=Error copying attributes -25202} /usr/local/lib/node_modules/appium/node_modules/appium-xcuitest-driver/WebDriverAgent/WebDriverAgentRunner/UITestingUITests.m 35 1
[Xcode] 2017-05-19 17:07:10.897 XCTRunner[25815:242886] Failed to fetch hit point for Button 0x608001160e40: traits: 8589934601, {{255.0, 423.0}, {100.0, 43.0}}, identifier: 'angelFollowButton', label: 'Leave' - *** -[__NSPlaceholderArray initWithObjects:count:]: attempt to insert nil object from objects[1]

## Link to Appium logs

Error at line 2472
https://gist.github.com/Rekubot/a27b43f528691c9f03695d6c687cc735#file-gistfile1-txt-L2472

Edit: I was hoping that only calling FindElement once and caching the button would solve the issue, but it seems to be the multiple calls to getText and getLocation before attempting to click it that causes the failure. Will investigate further on Monday if necessary.

**Comment by Rekubot** [comment]
This appears to be an issue with the latest Appium beta version. I've reverted back to 1.6.4 for now.

**Comment by Rekubot** [comment]
Using new TouchAction(driver).tap(element).perform() instead of element.click() gets around this issue even in the latest beta version.
